bigtop-tests/smoke-tests/hive/HiveJdbcGeneralTest.java

```diff
@@ -0,0 +1,390 @@
+      executeStatement(stmt,
+          "Create view testview AS select SUM(btest.boarded) BOARDED from btest, btestv where btest.Dest=btestv.Dest");
+      assertEquals("45072", printResults(stmt, "Select * from testview"));
+      executeStatement(stmt, "Drop view testview");
+      printResults(stmt, "Describe formatted btest");
+      setNegativeFetchSize(stmt);
+    }
+  }
+  @Test public void testTableDeletion() throws Exception{
+    try (Statement stmt = con.createStatement()) {
+      dropTable(stmt, newTableName);
+      dropTable(stmt, newTableName + "NT");
+      dropTable(stmt, newTableName + "T");
+      dropTable(stmt, newTableName + "P");
+      dropTable(stmt, newTableName + "V");
+    }
+  }
+
+  @Test
+  public void testFunctionsandMetaData() throws Exception {
+
+    String queryValues = "insert into table btestm values"
+        + " ('Alfred', 'M', 69.0, 122.5, 'AJH', 1),"
+        + " ('Alfred', 'M', 71.0, 130.5, 'AJH', 2),"
+        + " ('Alicia', 'F', 56.5, 84.0, 'BJH', 1),"
+        + " ('Alicia', 'F', 60.5, 86.9, 'BJH', 2),"
+        + " ('Benicia', 'F', 65.3, 98.0, 'BJH', 1),"
+        + " ('Benicia', 'F', 69.3, 99.1, 'BJH', 2),"
+        + " ('Bennett', 'F', 63.2, 96.2, 'AJH', 1),"
+        + " ('Bennett', 'F', 69.2, 98.2, 'AJH', 2),"
+        + " ('Carol', 'F', 62.8, 102.5, 'BJH', 1),"
+        + " ('Carol', 'F', 65.3, 105.4, 'BJH', 2),"
+        + " ('Carlos', 'M', 63.7, 102.9, 'AJH', 1),"
+        + " ('Carlos', 'M', 70.3, 106.9, 'AJH', 2),"
+        + " ('Henry', 'M', 63.5, 102.5, 'AJH', 1),"
+        + " ('Henry', 'M', 68.9, 108.6, 'AJH', 2),"
+        + " ('Jaime', 'M', 57.3, 86.0, 'BJH', 1),"
+        + " ('Jaime', 'M', 62.9, 90.0, 'BJH', 2),"
+        + " ('Janet', 'F', 59.8, 84.5, 'AJH', 1),"
+        + " ('Janet', 'F', 62.5, 86.5, 'AJH', 2),"
+        + " ('Jean', 'M', 68.2, 113.4, 'AJH', 1),"
+        + " ('Jean', 'M', 70.3, 116.0, 'AJH', 2),"
+        + " ('Joyce', 'M', 51.3, 50.5, 'BJH', 1),"
+        + " ('Joyce', 'M', 55.5, 53.5, 'BJH', 2),"
+        + " ('Luc', 'M', 66.3, 77.0, 'AJH', 1),"
+        + " ('Luc', 'M', 69.3, 82.9, 'AJH', 2),"
+        + " ('Marie', 'F', 66.5, 112.0, 'BJH', 1),"
+        + " ('Marie', 'F', 69.5, 114.9, 'BJH', 2),"
+        + " ('Medford', 'M', 64.9, 114.0, 'AJH', 1),"
+        + " ('Medford', 'M', NULL, NULL, NULL, NULL),"
+        + " ('Philip', 'M', 69.0, 115.0, 'AJH', 1),"
+        + " ('Philip', 'M', 70.0, 118.0, 'AJH', 2),"
+        + " ('Robert', 'M', 64.8, 128.0, 'BJH', 1),"
+        + " ('Robert', 'M', 68.3, NULL, 'BJH', 2),"
+        + " ('Thomas', 'M', 57.5, 85.0, 'AJH', 1),"
+        + " ('Thomas', 'M', 59.1, 92.3, 'AJH', 2),"
+        + " ('Wakana', 'F', 61.3, 99.0, 'AJH', 1),"
+        + " ('Wakana', 'F', 63.8, 102.9, 'AJH', 2),"
+        + " ('William', 'M', 66.5, 112.0, 'BJH', 1),"
+        + " ('William', 'M', 68.3, 118.2, 'BJH', 2)";
+
+    // failing query from SAS Institute, code generated by PROC MEANS
+    String failingQuery =

```

Comment:
What do you mean by failingQuery? Arn't we supposed to be testing positive cases here?

Reply:
Should be, but in this case the query is failing for some reason, I believe this to be a regression in Hive 3.0 Beta

Reply:
From Bigtop point of view we are testing against the release version, which is 2.3.3. If user want's to test other version of Hive, that's very much reasonable. 
In this example, the test should be passed in 2.3.3(I'm not sure though). If we expect it should be passed in 3.0 as well, then we should write the test as a normal one which assert that it should pass. 
Failing is another story that developer need to deal with, which don't need to encode in the test.
Do you think this is reasonable?

Reply:
You are correct, in 2.3.3 the test does pass correctly, I can ammend the comment to reference version 3.0 but leave the test case in. Using the current Bigtop Harness the test will pass, but when the release version becomes 3.0 I anticipate there will be a failure.

Reply:
Isn't that what we want? A failure raised telling us something different when we bump up the version.